src/lib/ngsi10/QueryContextRequest.h

```diff
@@ -56,10 +56,11 @@ typedef struct QueryContextRequest
   int               restrictions;
   StringList        metadataList;     // From URI param 'metadata'
   std::string       contextProvider;  // Not part of the payload - used internally only
+  std::string       forwardingMode;   // Not part of the payload - used internally only
 
   QueryContextRequest();
-  QueryContextRequest(const std::string& _contextProvider, EntityId* eP, const std::string& attributeName);
-  QueryContextRequest(const std::string& _contextProvider, EntityId* eP, const StringList& attributeList);
+  QueryContextRequest(const std::string& _contextProvider, EntityId* eP, const std::string& attributeName, const std::string& _forwardingMode);
+  QueryContextRequest(const std::string& _contextProvider, EntityId* eP, const StringList& attributeList, const std::string& forwardingMode);
```

Comment:
Why query context (a class that models the query context request itself) needs include aspects related with CPr format?

Reply:
I see the existing contextProvider there. So I think there should be a good reason but I don't remember it :)

Reply:
Wherever the provider is I needed to add a "providerFormat".
But, as I stated before, perhaps the providerFormat should be a part of the Provider?

Reply:
I understand the rationale ("given there is a provider-thing there, I'm addint this new one") but my question is more basic... why query context request class included a context provider to begin with (I mean, as it is now in master)?

Reply:
That I do not know.
We could look into it if you think it smells ...
Also, when making this change work, I "blindly added" providerFormat wherever I found a provider ...
This one might not be necessary. Let's look into it

Reply:
Still not 100% sure (it's a jungle ... :)) but I believe it is necessary.

Reply:
It seems it is used in queryForward() to associate the query request to the provider to which the request is forwareded, e.g:

```
  if (parseUrl(qcrP->contextProvider, ip, port, prefix, protocol) == false)
```

This fact means 100% security for me :)

NTC